### PR TITLE
Update raindrop-io extension

### PR DESCRIPTION
**Pull request by tdkn**
## Description

- feat: added ability to save currently opened browser tab
- refactor: some refactoring
- chore: npx ray migrate
- chore: update some packages

## Screencast

| 1 | 2 |
|--------|--------|
| Raycast 2024-05-19 at 08 49 55 | raindrop-io 2024-05-19 at 08 50 56 | 

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by tdkn** [comment]
I would like to add it later once I understand the rules for writing a CHANGELOG.md.

**Comment by tdkn** [comment]
Hmm, it looks like I need to update the screenshot under metadata as well.

**Comment by tdkn** [comment]
I think the pull request is ready for review 🙏

File: extensions/raindrop-io/src/hooks/useRequest.tsx
Comment (tdkn) [inline extensions/raindrop-io/src/hooks/useRequest.tsx:1]: Just renamed file extension:

```diff
-useRequest.tsx
+useRequest.ts
```

File: extensions/raindrop-io/src/hooks/useBrowserLink.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    async () => {
+      const app = await getFrontmostApplication();
+
+      switch (app.bundleId) {
+        case "company.thebrowser.Browser":
+          return runAppleScript(`tell application "Arc" to return URL of active tab of front window`);
+        case "com.google.Chrome":
+          return runAppleScript(`tell application "Google Chrome" to return URL of active tab of front window`);
+        case "com.apple.Safari":
+          return runAppleScript(`tell application "Safari" to return URL of front document`);
```
Comment (lardissone) [inline extensions/raindrop-io/src/hooks/useBrowserLink.ts:15]: Nice.

Can you add also Firefox?
I tried with this applescript and worked fine:
```
tell application "System Events" to get value of UI element 1 of combo box 1 of toolbar "Navigation" of first group of front window of application process "Firefox"
```

The bundleId seems to be `org.mozilla.firefox`:
```
<key>CFBundleIdentifier</key>
<string>org.mozilla.firefox</string>
```

**Review by sh-cho — APPROVED** [review]
Overall, looks good 👍

How about integrate auto completion in the existing "Add Bookmarks" command? Then, adding description about auto completion with [Form.Description](https://developers.raycast.com/api-reference/user-interface/form#form.description) or `info` property is enough I think.

File: extensions/raindrop-io/src/hooks/useBrowserLink.ts (hunk 1 above)
Comment (tdkn) [reply extensions/raindrop-io/src/hooks/useBrowserLink.ts:15]: @lardissone 
Thanks for the suggestion!
I wanna add support for Firefox too.
I tried adding the code, but got an error.
I'll check it out, but any clue what might be causing it?

```diff
diff --git a/src/hooks/useBrowserLink.ts b/src/hooks/useBrowserLink.ts
index b15f984..cedea6b 100644
--- a/src/hooks/useBrowserLink.ts
+++ b/src/hooks/useBrowserLink.ts
@@ -13,6 +13,10 @@ export function useBrowserLink() {
           return runAppleScript(`tell application "Google Chrome" to return URL of active tab of front window`);
         case "com.apple.Safari":
           return runAppleScript(`tell application "Safari" to return URL of front document`);
+        case "org.mozilla.firefox":
+          return runAppleScript(
+            `tell application "System Events" to get value of UI element 1 of combo box 1 of toolbar "Navigation" of first group of front window of application process "Firefox"`,
+          );
         default:
           throw new Error(`Unsupported App: ${app.name}`);
       }
```

raindrop-io 2024-05-21 at 00 06 25

CleanShot 2024-05-21 at 00 13 06@2x

Comment (lardissone) [reply extensions/raindrop-io/src/hooks/useBrowserLink.ts:15]: Weird, does it work if you call it from the terminal?

`osascript -e  'tell application "System Events" to get value of UI element 1 of combo box 1 of toolbar "Navigation" of first group of front window of application process "Firefox"'`:

20240520_121847@2x


I got the applescript from [this](https://apple.stackexchange.com/a/424375) SO post, maybe it requires to have voiceover disabled in Firefox? I have this value in `0` not `-1` as the post suggests.
20240520_121809@2x

Comment (tdkn) [reply extensions/raindrop-io/src/hooks/useBrowserLink.ts:15]: When experimenting with rewriting AppleScript to JXA, I don't know why, but after running `﻿SystemEvents.processes.byName("Firefox").properties()`, you get the URL of the current tab without an error.

I modified the AppleScript as follows. So far, this code seems to work well.

```AppleScript
tell application "System Events"
  set firefox to application process "Firefox"

  -- HACK:
  -- If you don't get the properties first, you'll get the error:
  -- `Can't get group 1 of window 1 of application process "Firefox". Invalid index. (-1719)`.
  get properties of firefox

  set frontWindow to front window of firefox
  set firstGroup to first group of frontWindow
  set navigation to toolbar "Navigation" of firstGroup
  get value of UI element 1 of combo box 1 of navigation
end tell
```

Comment (tdkn) [reply extensions/raindrop-io/src/hooks/useBrowserLink.ts:15]: @lardissone 
I've added a commit to the pull request, please check it out 😄

**Review by lardissone — APPROVED** [review]
lgtm! great work!

File: extensions/raindrop-io/src/hooks/useBrowserLink.ts (hunk 1 above)
Comment (lardissone) [reply extensions/raindrop-io/src/hooks/useBrowserLink.ts:15]: Good catch! Thanks!

**Comment by tdkn** [comment]
Hi, all.
I would also like to hear your opinions on @sh-cho's idea.

> How about integrate auto completion in the existing "Add Bookmarks" command? Then, adding description about auto completion with [Form.Description](https://developers.raycast.com/api-reference/user-interface/form#form.description) or info property is enough I think.

I initially thought the same thing, but I created it as a separate command so as not to change the behavior of the existing command.

**Comment by sh-cho** [comment]
@lardissone gentle ping :)

**Comment by lardissone** [comment]
@sh-cho ah yes, I approved it. Now it's in Raycast's hands.

/cc @pernielsentikaer

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by tdkn** [comment]
Your day seems to be filled with numerous pull requests.
I eagerly anticipate the day when this pull request is incorporated 🥹

**Comment by pernielsentikaer** [comment]
@tdkn, it looks like something went wrong in the merge part, triggering it again now 🙂

**Pull request merged**